### DISCUSSION TITLE
[Bug]: useAddonState: cannot read property 'getAddonState' of undefined in Storybook 7.5.1

### DISCUSSION THREAD

**Discussion by TriveniBrijesh** [storybookjs/storybook#25422] [Help]
### Describe the bug

Below is my manager.js file

import { addons } from "@storybook/preview-api";
import { useStorybookApi } from "@storybook/manager-api";
import * as DocBlock from "@storybook/blocks";

addons.register('MyAddon', () => {
  const api = useStorybookApi();
  console.log("api", api);
  const storyData = api.getCurrentStoryData();
  console.log("sd", storyData);
  api.setAddonContext("myAddon", storyData.importPath);
});

Below is my preview.js
import { useAddonState} from "@storybook/manager-api";

const preview = {
  parameters: {
    docs: {
      page: () => {
        const [myAddonState] = useAddonState('myAddon');
        console.log("addon", myAddonState);
              return (
          <>
            <DocBlock.Title />
            <DocBlock.Description />
            <DocBlock.Primary />
            <DocBlock.Controls />
            <DocBlock.Stories />
          </>
        )
      },
    },
    controls: {
      matchers: {
        color: /(background|color)$/i,
        date: /Date$/,
      },
    },
  },
};

export default preview;

### System

```bash
@storybook/react: 7.5.2
```

```[tasklist]
### Tasks
```

**Comment by vanessayuenn** [comment]
Hi, thanks for reporting this. do you have a reproduction repo you can share? If not, can you create one (see [how to create a repro](https://storybook.js.org/docs/react/contribute/how-to-reproduce)). We prioritize bug reports that have reproduction. Thank you! 🙏

**Comment by TriveniBrijesh** [comment]
@vanessayuenn 
I have created new sample repo and reproduced above issue.
https://github.com/brijesh231199/my-storybook-reproduction

**Comment by TriveniBrijesh** [comment]
@vanessayuenn 
Could you please provide an update? I have it as a priority.

**Comment by shilman** [comment]
@brijesh231199 you are using `useAddonState` from `@storybook/manager-api`. However Docs renders in the preview. Therefore you can only use hooks from `@storybook/preview-api` in your Docs code. What are you trying to achieve? Converting this to a discussion since it is a support request and not an actual bug in Storybook.

**Comment by TriveniBrijesh** [comment]
@shilman I have different changelog file for each component. I want to show changelog in docs file using <DocBlock.Markdown> element.
So how to get changelog file path in preview for every component docs.
